_posts/2020-01-24-inception-pipeline-cdk.md

```diff
@@ -0,0 +1,395 @@
+---
+layout: post
+title: Cross-Account Inception Pipeline with AWS CDK
+date: 2020-01-20
+tags: codepipeline cloud-development-kit infrastructure-as-code aws cdk cross-account
+author: JK Gunnink
+image: img/inception-pipelines/seed_germination.png
+---
+
+The [inception pipeline][inception-pipeline-post] which Mechanical Rock's very own Pete Yandell
+pioneered back in early 2018 was created as a way to manage a pipeline through code, which even has
+the ability to update itself through CloudFormation steps.
+
+Using this for a few projects over the past 18 months or so has highlighted its value as projects
+develop, more steps are added, additional permissions are required or locked down and once you have
+a few resources, it quickly becomes a very large YAML file with over 1,000 lines of text. Anyone
+who's ever written large CloudFormation files will understand the pains of making sure resources are
+lined up and will be created and updated in the way you expect. Yes there are linting tools and
```

Comment:
Why not nested stacks to fix this problem?

Reply:
Probably? Yes? Not without more overhead though managing the nested stack and passing references around etc.

Reply:
I think the big problem with CodePipeline in particular is less the size  and more the complications with KMS keys, permissions, bucket ownership, pass-roles, assume roles, etc etc.

I've not seen a single instance of a codepipeline template anywhere that does everything correctly that a) wasn't published by AWS, and/or b) wasn't limited to a single account,  and/or c) wasn't written via CDK.

Reply:
Resolved here ed2d5fd6c20cc6ec1ec98e35d85dd77d7ded7323